File: Python/removeDuplicates.py
def removeDuplicates(nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    if not nums:
        return 0
    '''
    nums = list(set(nums[:]))
    print nums 
    return len(nums)
    '''
    newTail = 0
    for i in xrange(1,len(nums)):
        if nums[i] != nums[newTail]:
            newTail += 1
            nums[newTail] = nums[i]
    print nums
    return newTail + 1
        
    

print removeDuplicates([1,1,2])

File: src/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    var i = 1, //iterate the array
        j = 0; // max index of different numbers
    if (nums.length < 2) return nums.length;
    for (; i < nums.length; i++) {
        // if two numbers are not same, then increase j and safe new number at index j; else just increase i and move right.
        if (nums[i] !== nums[i - 1]) {
            j++;
            nums[j] = nums[i];
        }
    }
    // as j is index, so return j+1 as lenght
    return j + 1;
};